client/app/states/marketplace/details/details.state.js

```diff
@@ -84,21 +85,34 @@
       return dialogFieldData;
     }
 
+    function alreadyInCart() {
+      return ShoppingCart.isDuplicate(dataForSubmit('service_template_href'));
+    }
+
     function addToCart() {
       if (!ShoppingCart.allowed()) {
         return;
       }
 
       var dialogFieldData = dataForSubmit('service_template_href');
 
+      vm.addingToCart = true;
+
       ShoppingCart.add({
         description: vm.serviceTemplate.name,
         data: dialogFieldData,
       })
-      .then(addSuccess, addFailure);
+      .then(addSuccess, addFailure)
+      .then(function() {
+        vm.addingToCart = false;
+      });
 
       function addSuccess(result) {
-        Notifications.success(__("Item added to shopping cart"));
+        if (result.duplicate) {
+          Notifications.success(__("Item added to shopping cart, but it's a duplicate of an existing item"));
```

Comment:
@AllenBW nope, this is a darga backport of #199. We don't have EventNoitification in darga ;)